outline/shadowsocks/config.go

```diff
@@ -35,16 +42,47 @@ type Client struct {
 	client.Client
 }
 
+func extractPrefixBytes(prefixUtf8 string) ([]byte, error) {
+	// The prefix is an 8-bit-clean byte sequence, stored in the codepoint
+	// values of a unicode string, which arrives here encoded in UTF-8.
+	prefixRunes := []rune(prefixUtf8)
+	prefixBytes := make([]byte, len(prefixRunes))
+	for i, r := range prefixRunes {
+		if (r & 0xFF) != r {
+			return nil, fmt.Errorf("character out of range: %d", r)
+		}
+		prefixBytes[i] = byte(r)
+	}
+	return prefixBytes, nil
+}
+
 // NewClient provides a gobind-compatible wrapper for [client.NewClient].
 func NewClient(config *Config) (*Client, error) {
```

Comment:
Expose some new `NewClient(configString)`, so it's easier to use.

Reply:
Make the old function call the new one so we can eventually delete it.

Reply:
> Expose some new `NewClient(configString)`, so it's easier to use.

Done.  Go doesn't support function overloading, and we're not breaking the existing API, so the new function is called `MakeClient`.

Reply:
Good point. The work around is to have the function name indicate the input differences.
I think NewClient and MakeClient is a bit confusing. Perhaps call the new one `NewClientFromJSON`? That way we keep the `NewClient` convention, and make it clear what the difference is.

In the future if we want to create a new client in yet another way, it scales better to find another suffix than a synonym of "make/new/create" that doesn't tell what the difference it.